src/routes/subscriptions/subscription.controller.spec.ts

```diff
@@ -0,0 +1,165 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { AppModule } from '@/app.module';
+import configuration from '@/config/entities/__tests__/configuration';
+import { EmailControllerModule } from '@/routes/email/email.controller.module';
+import { EmailApiModule } from '@/datasources/email-api/email-api.module';
+import { TestEmailApiModule } from '@/datasources/email-api/__tests__/test.email-api.module';
+import { EmailDataSourceModule } from '@/datasources/email/email.datasource.module';
+import { TestEmailDatasourceModule } from '@/datasources/email/__tests__/test.email.datasource.module';
+import { CacheModule } from '@/datasources/cache/cache.module';
+import { TestCacheModule } from '@/datasources/cache/__tests__/test.cache.module';
+import { RequestScopedLoggingModule } from '@/logging/logging.module';
+import { TestLoggingModule } from '@/logging/__tests__/test.logging.module';
+import { NetworkModule } from '@/datasources/network/network.module';
+import { TestNetworkModule } from '@/datasources/network/__tests__/test.network.module';
+import { IEmailDataSource } from '@/domain/interfaces/email.datasource.interface';
+import { TestAppProvider } from '@/__tests__/test-app.provider';
+import * as request from 'supertest';
+import { faker } from '@faker-js/faker';
+import { Subscription } from '@/domain/email/entities/subscription.entity';
+
+describe('Subscription Controller tests', () => {
+  let app;
+  let emailDatasource;
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    jest.useFakeTimers();
+
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule.register(configuration), EmailControllerModule],
+    })
+      .overrideModule(EmailApiModule)
+      .useModule(TestEmailApiModule)
+      .overrideModule(EmailDataSourceModule)
+      .useModule(TestEmailDatasourceModule)
+      .overrideModule(CacheModule)
+      .useModule(TestCacheModule)
+      .overrideModule(RequestScopedLoggingModule)
+      .useModule(TestLoggingModule)
+      .overrideModule(NetworkModule)
+      .useModule(TestNetworkModule)
+      .compile();
+
+    emailDatasource = moduleFixture.get(IEmailDataSource);
+
+    app = await new TestAppProvider().provide(moduleFixture);
+    await app.init();
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  it('deletes category successfully', async () => {
```

Comment:
Very nitpick, but: I'd say `unsubscribes from a caategory` or similiar, instead of `deletes category` (also in the other test cases). Otherwise I am under the impresion the test verifies the removal of a category itself.

Also, pretty unrelated to thi PR: the endpoing uses the keyword `category`, which I think it's meaningful, but at the database/datasource level we use `subscription` to refer to a category/topic. Does it make sense to unify this semantically?

Reply:
Done in [e075ea0](https://github.com/safe-global/safe-client-gateway/pull/1037/commits/e075ea00852417361e6628660ca91308a1d25852)